### ISSUE TITLE
A question about training

### ISSUE THREAD

**Issue by l-cr** [FightingFighting/GPS#6]
Hi!
I found that the model parameters are not frozen during the training process; instead, only selected updates to the weight matrix parameters are made using a mask. The memory and time required for this are not significantly different from full fine-tuning. Are there any methods to improve this?

**Comment by FightingFighting** [comment]
Hi, It is true that the masking implementation way results in more memory during training. The memory reduction in Lora forms the reduction of trainable parameters in the optimizer . So there is a possible way to implement it. For example, remember the position of the selected parameters and only store these parameters in optimizer instead of all parameters using mask.

**Comment by l-cr** [comment]
> Hi, It is true that the masking implementation way results in more memory during training. The memory reduction in Lora forms the reduction of trainable parameters in the optimizer . So there is a possible way to implement it. For example, remember the position of the selected parameters and only store these parameters in optimizer instead of all parameters using mask.

The training aims at the connection parameters between neurons, but the optimizer targets the entire weight matrix. I think it's difficult to implement. If you have any ideas, please provide a code example.

**Comment by FightingFighting** [comment]
Hi @l-cr sorry, for this version, we cannot provide it. It might be provided in the future version.

**Comment by l-cr** [comment]
> Hi @l-cr sorry, for this version, we cannot provide it. It might be provided in the future version.

Thank you, and I hope you can go further.
